Tools/GNUMake/comps/gnu.mak

```diff
@@ -114,12 +114,8 @@ endif
 
 ########################################################################
 
-ifeq ($(gcc_major_version),4)
```

Comment:
If we make this change, it seems appropriate to me for the makefile to throw an error if the user attempts to use C++14 with gcc 4.x, for precisely the reason you're saying about avoiding incomprehensible warnings.

Reply:
Sounds good, yes!

Reply:
Hm, on second though: we actually have to do <= and >= comparisons here for major versions and C++ std version strings which are not first-class citizens in GNUMake. Not sure how much of CMake's compiler sanity checks we want to re-implement.

Reply:
But can do shell calls if that's ok. `$(shell expr $(CXXSTD) \> 11)`, etc.

Reply:
We don't have to enforce every sanity check, but there is a specific historical reason why we had that particular check and it might be good to keep it. It's unlikely that anyone is using gcc <4, so the check as it currently is seems fine to me.

Reply:
Alright, updated anyway and tested now.